lib/modules/datasource/crate/index.ts

```diff
@@ -362,6 +348,56 @@ export class CrateDatasource extends Datasource {
     return registry;
   }
 
+  private static async clone(
+    registryFetchUrl: string,
+    clonePath: string,
+    packageName: string,
+    cacheKeyForError: string,
+  ): Promise<string> {
+    const git = Git({
+      ...simpleGitConfig(),
+      maxConcurrentProcesses: 1,
+    }).env(getChildEnv());
+
+    const innerClonePromise = git.clone(registryFetchUrl, clonePath, {
+      '--depth': 1,
+    });
```

Comment:
It's been a few years since I played around with raw promises but I'm pretty sure that this line will already _kick off_ the clone. It's not waiting for you to await it. What's your rational of separating out the promise from the await?

Reply:
No rational. I'm mostly a Rust programmer where promises don't do anything unless you await them.

I did it this way, because the original code did it this way (to be able to cache the promise). But that's not needed here, so I'll rewrite it.

Reply:
Let me take another look first, thanks